### PR TITLE
vectorTile to geojson directly in 'EPSG:3857'

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description
vectorTile to geojson directly in 'EPSG:3857'

### CODE REVIEW

File: examples/globe_vector_tiles.html [hunk 1]
```diff
@@ -40,30 +40,32 @@
...
                 var backgroundLayer;
 
                 style.layers.forEach(function(layer) {
                     if (layer.type === 'background') {
                         backgroundLayer = layer;
-                    } else if (['fill', 'line'].indexOf(layer.type) >= 0) {
+                    } else if (['fill', 'line'].indexOf(layer.type) >= 0 &&
+                        ['landcover', 'water', 'boundary', 'transportation', 'park'].indexOf(layer['source-layer']) >= 0 &&
+                        layer.id.indexOf('bridge') < 0 &&
+                        layer.id.indexOf('bridge') < 0 &&
```
Comment (zarov) [inline examples/globe_vector_tiles.html]: Double to remove

File: examples/globe_vector_tiles.html [hunk 2]
```diff
@@ -40,30 +40,32 @@
...
+                        layer.id.indexOf('tunnel') < 0 &&
+                        layer.id.indexOf('admin_sub') < 0) {
                         supportedLayers.push(layer);
                     }
                 });
 
                 function isValidData(data, extentDestination) {
-                    return extentDestination.zoom - data.extent.zoom < 4;
+                    var z = extentDestination.zoom - 2;
+                    return extentDestination.zoom, z - z % 4 == data.extent.zoom - 2;
```
Comment (zarov) [inline examples/globe_vector_tiles.html:65]: What is this check ?

File: src/Parser/VectorTileParser.js [hunk 3]
```diff
@@ -2,6 +2,110 @@ import Protobuf from 'pbf';
...
+const VectorTileFeature = { types: ['Unknown', 'Point', 'LineString', 'Polygon'] };
+const s = 20037508.34;
+function project(line, ox, oy, size) {
+    for (let j = 0; j < line.length; j++) {
+        const p = line[j];
+        line[j] = [((p.x + ox) / size - 0.5) * s * 2, (0.5 - (p.y + oy) / size) * s * 2];
+    }
+}
+
+function toGeoJSON(x, y, z) {
```
Comment (zarov) [inline src/Parser/VectorTileParser.js:61]: Please specify that this is freely adapted from `vector-tile-js`

**Review by zarov — CHANGES_REQUESTED** [review]
Thanks for these changes, that's way better ! There are some things to polish and that'll be good

**Comment by gchoqueux** [comment]
@zarov  PR updated

**Review by zarov — APPROVED** [review]
Thanks for the changes

**Pull request merged**